x/dymns/keeper/hooks.go

```diff
@@ -0,0 +1,370 @@
+				return err
+			}
+
+			return nil
+		})
+
+		if errApplyStateChange == nil {
+			aSoe.Remove(so.AssetId)
+		}
+	}
+
+	if err := e.SetActiveSellOrdersExpiration(ctx, aSoe, dymnstypes.TypeAlias); err != nil {
+		e.Keeper.Logger(ctx).Error("failed to update active SO expiry", "error", err)
+		return err
+	}
+
+	return nil
+}
+
+// getFinishedSellOrders returns the finished Sell-Orders for the asset type.
+// Finished Sell-Orders are the Sell-Orders that have expired or completed by having a bid.
+func (e epochHooks) getFinishedSellOrders(ctx sdk.Context, assetType dymnstypes.AssetType, logger log.Logger) (
+	finishedSellOrders []dymnstypes.SellOrder,
+	aSoe *dymnstypes.ActiveSellOrdersExpiration,
+) {
+	aSoe = e.Keeper.GetActiveSellOrdersExpiration(ctx, assetType)
+	if len(aSoe.Records) < 1 {
+		return
+	}
+
+	blockEpochUTC := ctx.BlockTime().Unix()
+	logger = logger.With("asset-type", assetType.FriendlyString(), "time", blockEpochUTC)
+
+	/*
+		The following part is to filter out the finished Sell-Orders,
+		plus one logic to correct the invalid Sell-Order expiry.
+		Currently, there is no such case that can cause an invalid Sell-Order expiry,
+		but it's good to have a correction logic for the future
+		because this part is executed by hook, and to make sure the hook always be executed efficiently,
+		we need to make sure the data is always valid.
+	*/
+	invalidRecordsToRemove := make([]string, 0)
+
+	for i, record := range aSoe.Records {
+		if record.ExpireAt > blockEpochUTC {
+			// skip not expired ones
+			continue
+		}
+
+		so := e.GetSellOrder(ctx, record.AssetId, assetType)
+
+		if so == nil {
+			// remove the invalid entry
+			invalidRecordsToRemove = append(invalidRecordsToRemove, record.AssetId)
+			continue
+		}
+
+		if !so.HasFinished(blockEpochUTC) {
+			// invalid entry
+			logger.Error("sell order has not finished", "asset-id", record.AssetId, "expiry", record.ExpireAt)
+
+			aSoe.Records[i].ExpireAt = so.ExpireAt // correct it
+			continue
+		}
+
+		finishedSellOrders = append(finishedSellOrders, *so)
+	}
+
+	for _, name := range invalidRecordsToRemove {
+		aSoe.Remove(name)
+	}
+
+	return
+}
+
+/* -------------------------------------------------------------------------- */
+/*                             x/rollapp hooks                                */
+/* -------------------------------------------------------------------------- */
+
+// GetRollAppHooks returns the RollApp hooks struct.
+func (k Keeper) GetRollAppHooks() rollapptypes.RollappHooks {
+	return rollappHooks{
+		Keeper: k,
+	}
+}
+
+type rollappHooks struct {
+	Keeper
+}
+
+var _ rollapptypes.RollappHooks = rollappHooks{}
+
+func (h rollappHooks) RollappCreated(ctx sdk.Context, rollappID, alias string, creatorAddr sdk.AccAddress) error {
+	if alias == "" {
+		return nil
+	}
+
+	// ensure RollApp record is set
+	if !h.Keeper.IsRollAppId(ctx, rollappID) {
+		return errorsmod.Wrapf(gerrc.ErrInvalidArgument, "not a RollApp chain-id: %s", rollappID)
+	}
```

Comment:
why? can't you trust the caller?

Reply:
That's future handling, code can be changed, developers may make mistake.
When you can use a line of code, handle future potential issues, not allowing something surprise, how many resource needed to take care for the issue happens? So why not?